### DISCUSSION TITLE
Setup a button to open a specific software

### DISCUSSION THREAD

**Discussion by ericsacchetto** [NicoHood/HID#276] [Q&A]
Hi there,

I am trying to implement a button to open a specific software in Windows.

So, pressing a button would open Steam, for example.
C:\Program Files (x86)\Steam\Steam.exe

I am not sure if that is possible or how to do it. Can anyone help me?

Thanks

**Comment by chechu2k5** [comment]
1. Map your hardware button to generate one of the extended [Fn keys](https://github.com/NicoHood/HID/blob/e0912bab974b347f44e88c22d95fa7950511cda9/src/KeyboardLayouts/ImprovedKeylayouts.h#L176)  
2. use an [Autohotkey](https://www.autohotkey.com/) script to launch the steam exe like so
```autohotkey
; press of F13
F13::Run "C:\Program Files (x86)\Steam\Steam.exe"
return

; press of Shift+F15 will run notepad
+F15::Run "notepad.exe"
return

; press of Win+F14 will send the following text verbatim to the window with focus
#F14:: SendInput {Text}Foobar1000
return

; press of Alt+F24 will kill the process whose window is having focus (that is possibly not responding)
!F24::
    WinGet, PID, PID, % "ahk_id " WinExist("A")
    Process, Close, %PID%
return
```

> Note: Most but not all combinations of `F13` through `F24` and modifiers `Ctrl, Shift, Alt, Win` can be used as triggers.
> One unusable combination I know of is `Win+F16` which triggers a shutdown sequence on Windows 10.
